wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -888,24 +888,29 @@ public final class SessionManager: NSObject, SessionManagerType {
         }
     }
 
-    private func switchAnalyticsUser() {
+    private func switchAnalyticsUser(for userSession: ZMUserSession) {
         guard
             let analyticsManager,
-            let activeUserSession
+            let userProfile = getUserAnalyticsProfile(for: userSession)
         else {
             return
```

Comment:
Is this a valid state? 

If not I think you should log the error and do an `assertionFailure()` to surface this issue, especially since there are a lot of possible ways `getUserAnalyticsProfile(for:)` can return nil

Reply:
It is a valid state in some cases analytics are disabled etc.